### PR TITLE
[bugfix](scanner) Fix scanner of topN of duplicate key with scanner pooling

### PR DESCRIPTION
**Pull request by zclllyybb**
# Proposed changes

Issue Number: close DORIS-4362

## Problem summary

before:
```
mysql> SELECT ss_item_sk FROM store_sales ORDER BY ss_sold_date_sk DESC, ss_sold_time_sk DESC, ss_item_sk DESC LIMIT 10;
+------------+
| ss_item_sk |
+------------+
|      17899 |
|      17899 |
|      16866 |
|      16866 |
|      16236 |
|      16236 |
|      16161 |
|      16161 |
|      15012 |
|      15012 |
+------------+
```
after:
```
mysql> SELECT ss_item_sk FROM store_sales ORDER BY ss_sold_date_sk DESC, ss_sold_time_sk DESC, ss_item_sk DESC LIMIT 10;
+------------+
| ss_item_sk |
+------------+
|      17899 |
|      16866 |
|      16236 |
|      16161 |
|      15012 |
|      11905 |
|       9205 |
|       8280 |
|       6345 |
|       5400 |
+------------+
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

### CODE REVIEW

File: be/src/exec/olap_utils.h [hunk 1]
```diff
@@ -49,6 +53,15 @@ struct OlapScanRange {
     bool end_include;
     OlapTuple begin_scan_range;
     OlapTuple end_scan_range;
+
+    std::string debug_string() const {
+        std::stringstream ss;
```
Comment (HappenLee) [inline be/src/exec/olap_utils.h:58]: use fmt buffer, do not use stringstream.

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.28 seconds
 stream load tsv:          432 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230512140938_clickbench_pr_143020.html

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**